lib/services/user_service.ex

```diff
@@ -12,6 +12,16 @@ defmodule SlackCoder.Services.UserService do
     if db_user, do: {:ok, user}, else: {:new, user}
   end
 
+  def save(user, changeset) do
+    if changeset.changes[:slack] || changeset.changes[:github] do
+      SlackCoder.Users.Supervisor.stop_user(user)
+      user = Repo.update(changeset)
+      SlackCoder.Users.Supervisor.start_user(user)
+    else
+      SlackCoder.Users.Supervisor.start_user(user)
```

Comment:
You should be able to always call start user, since it should find it if its already started:
https://github.com/mgwidmann/slack_coder/blob/master/lib/slack_coder/users/supervisor.ex#L22